File: A Balanced Problemset?.py
def solve():
    n, x = map(int, input().split())
    ans = 1
    for i in range(1, int(n**0.5) + 1):
        if n % i == 0:
            z1 = n // i
            if z1 >= x:
                ans = max(ans, i)
            if i >= x:
                ans = max(ans, z1)
    print(ans)

def main():
    t = int(input())
    for _ in range(t):
        solve()

if __name__ == "__main__":
    main()

File: A Balanced Problemset?.cpp
#include <bits/stdc++.h>
using namespace std;
void solve()
{
   int n,x;
   int ans=1;
   cin>>n>>x;
   for(int i=1;i*i<=n;i++)
   {
      if(n%i==0)
      {
      int z1=(n/i);
      if(z1>=x)
      {
	 ans=max(ans,i);
      }
      if(i>=x)
      {
         ans=max(ans,z1);
      }
      }
   }
   cout<<ans<<endl;
   return;
}

 
int main() {
 
    int t;
    cin >> t;
    
    while (t--) 
    {
        solve();
    }
    return 0;
}
